cmd/checkpoint/main.go

```diff
@@ -486,47 +460,49 @@ func getAPIParentPods(client kubernetes.Interface, nodeName string) map[string]*
 // A minimal kubelet client. It assumes the kubelet can be reached the kubelet's insecure API at
 // 127.0.0.1:10255 and the secure API at 127.0.0.1:10250.
 type kubeletClient struct {
-	client *http.Client
+	insecureClient *rest.RESTClient
+	secureClient   *rest.RESTClient
 }
 
-// localParentPods will retrieve all pods from kubelet api that are parents & should be checkpointed
-func (c *kubeletClient) localParentPods() (map[string]*v1.Pod, error) {
-	resp, err := c.client.Get(kubeletAPIPodsURL)
-	if err != nil {
-		return nil, fmt.Errorf("failed to contact kubelet pod api: %v", err)
-	}
+func newKubeletClient(config *rest.Config) (*kubeletClient, error) {
+	// Shallow copy.
+	insecureConfig := *config
+	secureConfig := *config
 
-	defer resp.Body.Close()
-	b, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return nil, fmt.Errorf("failed to read response: %v", err)
-	}
+	insecureConfig.Host = "http://127.0.0.1:10255"
+	secureConfig.Host = "https://127.0.0.1:10250"
+	// Kubelet is using a self-signed cert.
+	secureConfig.TLSClientConfig.Insecure = true
 
-	var podList v1.PodList
-	if err := json.Unmarshal(b, &podList); err != nil {
-		return nil, fmt.Errorf("failed to unmarshal podlist: %v", err)
+	client := new(kubeletClient)
+	var err error
+	if client.insecureClient, err = rest.RESTClientFor(&insecureConfig); err != nil {
+		return nil, fmt.Errorf("failed creating kubelet client: %v", err)
 	}
-	return podListToParentPods(&podList), nil
-}
-
-// localRunningPods uses the /runningpods/ kubelet api to retrieve the local container runtime pod state
-func (c *kubeletClient) localRunningPods() (map[string]*v1.Pod, error) {
-	resp, err := c.client.Get(kubeletAPIRunningPodsURL)
-	if err != nil {
-		return nil, fmt.Errorf("failed to contact kubelet pod api: %v", err)
+	if client.secureClient, err = rest.RESTClientFor(&secureConfig); err != nil {
+		return nil, fmt.Errorf("failed creating kubelet client: %v", err)
```

Comment:
I don't think the user actually cares about the difference between these two. Let me improve the docstrings on the kubeletClient struct explaining why we have an insecure client and a secure client.